documentation/prover-incentives.md

```diff
@@ -0,0 +1,69 @@
+
+## The problem
+
+Verifying proofs on the Ethereum L1 is a [costly operation](https://docs.alignedlayer.com/#why-are-we-building-aligned) in terms of gas. Groth16 proofs, which are some of the cheapest to verify cost around 250k gas, while STARKS cost over 1M gas. Because of this, most zk rollups call the verification function of their inbox contract only sporadically.  
+
+We would like to get a similar property for validity Based rollups - but that opens the question, who should verify the batches submitted to the L1 and how can we incentivize proposers to submit valid batches if they don’t get proven immediately? A common approach to this is to force proposers to deposit a bond when they submit a batch, and force them to provide a validity proof within a time window, otherwise they lose their stake and proving becomes permisionless. This is the approach deployed on [Taiko](https://docs.taiko.xyz/resources/terminology/#validity-bond) today.
+
+- After the pacaya fork, there is now an option to prove multiple non consecutive batches(and anyone can do it), but if the batch is within its proving window(60 minutes as of this writing) the rewards goes to the proposer.
+This is a bit more flexible, but the incentives are basically the same - each proposer is incentivized to prove or get a prover for their batches only, which does not result in the most efficient proving costs. If the proposers are sophisticated enough, an off-chain market may develop, where different proposers pay the same prover to generate a proof for their batch, and the prover may aggregate those proofs in order to get a more efficient result. This is still uncertain, and it may even have a very centralizing effect, since provers are incentivized to make a deal with as many proposers as possible to ensure they can prove consecutive batches and get paid off-chain(if different batches have proposers that have chosen different provers then none of them will be incentivized to prove the entire set).
+- This also has the downside that you have to store multiple checkpoints in the Inbox contract, because proving may happen out of order.
+
+## Proposed design
+
+Anyone that wants to be a prover can register as the designated prover by signaling how much they are willing to charge **on a per publication basis**. The prover that offers the lower amount gets monopoly rights to prove publications until someone else comes in and offers a lower cost per publication. This way proposers know exactly how much they will have to pay to get their publication proven, and are not required to put additional capital, making the system more capital efficient.  
+New participants that want to undercut the current prover need to offer a lower price per publication that **is at least a defined percentage lower** than the current prover or other participants. This is to avoid gas wars common in auctions where the new participant just offers one wei less than the current best offer.  
+
```

Comment:
We're missing a section here that details better how the undercutting percentage is agreed or calculated. I was thinking 0.1 ETH as (iirc) more than 99% of the L2 transactions are worth less than that.

Reply:
Ah right, I got messed up between two concepts. On one side there's the need of documenting the mechanics of the percentage work, and on the other we need a default fee value for each `period`. I see there's an [`initialFee` in the `ProverManager` constructor](https://github.com/OpenZeppelin/minimal-rollup/pull/49/files#diff-d43081579a98c4c833bda7e2444609810a56b9fb193513d8f66eed2232a113a5R114).

It's true the design document may not specify any implementation details, so I'm reconsidering whether we should document it here.

For both things, I the framework to get the minimal-rollup up and running should have defaults, and we will document tradeoffs.